engine/src/engine.rs

```diff
@@ -709,11 +723,11 @@ impl<'env, I: IO + Copy, E: Env> Engine<'env, I, E> {
             let mut message = args.msg.as_bytes();
             assert_or_finish!(message.len() >= 40, output_on_fail, self.io);
 
-            let recipient = Address(unwrap_res_or_finish!(
+            let recipient = ADDRESS(H160(unwrap_res_or_finish!(
```

Comment:
It seems likely to me that this expression could be simplified, but it's hard to tell because of the macro.